odc/stats/tasks.py

```diff
@@ -544,14 +543,22 @@ def stream_from_sqs(
             token = SQSWorkToken(msg, visibility_timeout)
             tidx, filedb = parse_sqs(msg.body)
 
-            # avoid the download and update again, how ever, we have to setup an exception to handle the unit test filedb
+            # Avoid downloading the file multiple times.
             if urlparse(filedb).scheme == 's3':
                 bucket, key = s3_url_parse(filedb)
-            else: # if it is the test_tiles.db, we load it from local
+            else:
+                # If it is the test_tiles.db, we load it from local
                 key = filedb
             local_cache_file = key.split("/")[-1]
-            if not os.path.isfile(local_cache_file):  # use the download filedb from S3 as the init context flag
+            # Use the downloaded filedb from S3 as the init context flag
+            if not os.path.isfile(local_cache_file):
                 self.init_from_sqs(filedb)
+            else:
```

Comment:
My original fix plan is: 

No matter we have the db_cache file locally or not, we will always run `self.init_from_sqs(filedb)`. But we do the if-else check on this https://github.com/opendatacube/odc-stats/blob/develop/odc/stats/tasks.py#L383 to not download same db_cache file twice.

Of course, your version also works.

Reply:
I think mine is simpler... but yours updates those other fields, so, I think yours is safer.

I'll change this PR.